src/Altinn.AccessManagement.Core/Services/Interfaces/ISingleRightsService.cs

```diff
@@ -47,6 +48,6 @@ public interface ISingleRightsService
         /// <param name="delegation">delegation</param>
         /// <param name="cancellationToken">http context token</param>
         /// <returns>The result of the deletion</returns>
-        Task<DelegationActionResult> RevokeRightsDelegation(int authenticatedUserId, DelegationLookup delegation, CancellationToken cancellationToken);
+        Task<ValidationProblemDetails> RevokeRightsDelegation(int authenticatedUserId, DelegationLookup delegation, CancellationToken cancellationToken);
```

Comment:
It kinda breaks the clean architecture principle behind the API/Core/Integrations principle if the core business logic starts to use API layer specific responses. Which is why we so far have opted for wrapper models and mapping in the API layer. 

The alternative being to throw exceptions and handle/map those is the API. But this has some additional overhead just for responding with a validation error, and lacks the support of resolving and responding with multiple validation errors.

Reply:
I agree that we should only return data / models from the service layer. I also prefer you alternative; throw a validation exception that can be transformed in the ExpcetionHandler middleware to a ProblemDetail. I really want to add this feature in another PR though and then fix these methods.